packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationModule.java

```
@@ -64,9 +68,15 @@ public void initialize() {
     }
 
     @ReactMethod
-    public void getToken(Callback callback) {
-        String token =  FirebaseInstanceId.getInstance().getToken();
-        Log.i(LOG_TAG, "getting token" + token);
-        callback.invoke(token);
+    public void getToken(final Callback callback) {
+        final Task<String> taskToken =  FirebaseMessaging.getInstance().getToken();
+			taskToken.addOnSuccessListener(new OnSuccessListener<String>() {
```

What happens if getToken fails?

---

With this current state. It silently fails. The previous code didn't handle it as well so I left as is. But I'll add it now. Thank you for catching that!